packages/data-mate/src/transforms/field-transform.ts

```diff
@@ -492,37 +1073,135 @@ export function toUnixTime(input: any, { ms = false } = {}) {
     return time as number;
 }
 
-export function toISO8601(input: any, args?: { resolution?: 'seconds' | 'milliseconds' }) {
+// option to specify, seconds, millisecond, microseconds?
+/**
+ * Converts a given date to its time in milliseconds or seconds
+ *
+ * @example
+ *
+ * expect(fieldTransform.toUnixTime('2020-01-01')).toBe(1577836800);
+ * expect(fieldTransform.toUnixTime('Jan 1, 2020 UTC')).toBe(1577836800);
+ * expect(fieldTransform.toUnixTime('2020 Jan, 1 UTC')).toBe(1577836800);
+ *
+ * expect(fieldTransform.toUnixTime(1580418907000)).toBe(1580418907);
+ * expect(fieldTransform.toUnixTime(1580418907000, { ms: true })).toBe(1580418907000);
+ *
+ * @export
+ * @param {*} input
+ * @param {*} [{ ms = false }={}] set ms to true if you want time in milliseconds
+ * @returns { number | number[] | null} returns null if input is null/undefined
+ */
+
+export function toUnixTime(input: any, { ms = false } = {}) {
     if (ts.isNil(input)) return null;
-    if (!isValidDate(input)) {
-        throw new Error('Not a valid date');
+
+    if (isArray(input)) {
+        return input.filter(ts.isNotNil).map((data: any) => makeUnitTime(data, { ms }));
     }
 
+    if (!isValidDate(input)) throw new Error('Not a valid date, cannot transform to unix time');
+
+    return makeUnitTime(input, { ms });
+}
+
+function makeIso(input: any, args?: { resolution?: 'seconds' | 'milliseconds' }) {
```

Comment:
We should be consistent with the naming here:

```suggestion
function _makeISO(input: any, args?: { resolution?: 'seconds' | 'milliseconds' }) {
```